qiskit/circuit/library/data_preparation/state_preparation.py

```diff
@@ -243,6 +243,20 @@ def validate_parameter(self, parameter):
     def _return_repeat(self, exponent: float) -> "Gate":
         return Gate(name=f"{self.name}*{exponent}", num_qubits=self.num_qubits, params=[])
 
+    ### helper functions for initializer.gates_to_uncompute() method ###
```

Comment:
What do you think about just implementing this on `Initialize` directly? It's not used in this class and we might accidentally remove it in future again, so it would be safer to just have it in the API-documented method.

Reply:
done in [b6f208a](https://github.com/Qiskit/qiskit/pull/12976/commits/b6f208acc968106349e4f5ef8398f343633db83a)

Reply:
I also added a test that will hopefully prevent us from breaking the API in the future